core/src/main/scala/codesearch/core/util/Helper.scala

```diff
@@ -40,47 +40,30 @@ object Helper {
     IO(Source.fromFile(path, "UTF-8")).bracket(source => IO.pure(source.getLines.toList))(source => IO(source.close()))
 
   def preciseMatch(query: String): String = {
-    val queryTokens: Seq[Token] = Tokenizer.parseStringWithSpecialSymbols(query)
+    val queryTokens: Seq[Token]  = Tokenizer.parseStringWithSpecialSymbols(query)
+    val specialChars: Set[String] = Set("|", "^", "$", "$", "+", "*", "(", ")", ".", "?")
     val preciseMatch: Seq[Token] = queryTokens.map {
-      case SpecialSymbol(value @ "|") => Escaped(value.charAt(0))
-      case SpecialSymbol(value @ "^") => Escaped(value.charAt(0))
-      case SpecialSymbol(value @ "$") => Escaped(value.charAt(0))
-      case SpecialSymbol(value @ "+") => Escaped(value.charAt(0))
-      case SpecialSymbol(value @ "*") => Escaped(value.charAt(0))
-      case SpecialSymbol(value @ "(") => Escaped(value.charAt(0))
-      case SpecialSymbol(value @ ")") => Escaped(value.charAt(0))
-      case SpecialSymbol(value @ ".") => Escaped(value.charAt(0))
-      case SpecialSymbol(value @ "?") => Escaped(value.charAt(0))
-      case other @ _                  => other
+      case SpecialSymbol(value) if specialChars.contains(value) => Escaped(value.charAt(0))
```

Comment:
Before it was: `case SpecialSymbol(value @ "|") => Escaped(value.charAt(0))`. Is it correct now?

Reply:
Yes.